services/src/main/java/org/keycloak/authorization/authorization/AuthorizationTokenService.java

```diff
@@ -636,12 +668,14 @@ private void resolveResourcePermission(KeycloakAuthorizationRequest request,
         }
 
         if (permissionsToEvaluate.isEmpty()) {
-            throw new CorsErrorResponseException(request.getCors(), "invalid_resource", "Resource with id [" + resourceId + "] does not exist.", Status.BAD_REQUEST);
+            CorsErrorResponseException invalidResourceException = new CorsErrorResponseException(request.getCors(), "invalid_resource", "Resource with id [" + resourceId + "] does not exist.", Status.BAD_REQUEST);
+            fireErrorEvent(request.getEvent(), Errors.INVALID_RESOURCE, invalidResourceException);
```

Comment:
I would just use Errors.INVALID_REQUEST. The details should be enough to understand what happened.

Reply:
agreed